File: src/main/java/dev/salmon/seraph/listener/event/LocrawEvent.java
package dev.salmon.seraph.listener.event;

import dev.salmon.seraph.util.locraw.LocrawInfo;
import net.minecraftforge.fml.common.eventhandler.Event;

public class LocrawEvent extends Event {
    public final LocrawInfo locraw;

    public LocrawEvent(LocrawInfo locraw) {
        this.locraw = locraw;
    }

    public static class JoinGame extends LocrawEvent {
        public JoinGame(LocrawInfo locraw) {
            super(locraw);
        }
    }

    public static class JoinLobby extends LocrawEvent {
        public JoinLobby(LocrawInfo locraw) {
            super(locraw);
        }
    }
}

File: src/main/kotlin/com/github/scherso/seraph/event/LocrawEvent.kt
package com.github.scherso.seraph.event

import club.maxstats.weave.loader.api.event.Event
import com.github.scherso.seraph.util.location.LocrawInfo

/**
 * @author Scherso ([...](https://github.com/Scherso))
 * @see Event
 */
open class LocrawEvent(val locraw: LocrawInfo) : Event() {

    /**
     * Posts from [com.github.scherso.seraph.util.location.LocrawUtils.onChatReceivedEvent]
     * on the event that a player joins a game.
     *
     * @param locraw Location Raw, the location of the player.
     * @see com.github.scherso.seraph.util.location.LocrawUtils
     */
    class JoinGame(locraw: LocrawInfo) : LocrawEvent(locraw)

    /**
     * Posts from [com.github.scherso.seraph.util.location.LocrawUtils.onChatReceivedEvent]
     * on the event that a player joins a lobby.
     *
     * @param locraw Location Raw, the location of the player.
     * @see com.github.scherso.seraph.util.location.LocrawUtils
     */
    class JoinLobby(locraw: LocrawInfo) : LocrawEvent(locraw)
}
